core/templates/dev/head/domain/collection/SkillListObjectFactory.js

```diff
@@ -0,0 +1,146 @@
+    // contained in the list.
+    SkillList.prototype.addSkill = function(skillName) {
+      if (this.containsSkill(skillName)) {
+        return false;
+      }
+      this.skillsList.push(skillName);
+      return true;
+    };
+
+    // Adds many skills. Returns false if not every skill was successfully
+    // added. See addSkill() for criteria on why individual skill adds may fail.
+    SkillList.prototype.addSkills = function(skillNames) {
+      var succeeded = true;
+      for (var i = 0; i < skillNames.length; i++) {
+        succeeded &= this.addSkill(skillNames[i]);
+      }
+      return succeeded;
+    };
+
+    // Resets the list to the given array of skill names. This is equivalent to
+    // clearing the list and then adding all of the skills again.
+    SkillList.prototype.setSkills = function(skillNames) {
+      this.clearSkills();
+      this.addSkills(skillNames);
+    };
+
+    // Returns the skill at the given index, or null if the index is out of
+    // bounds.
+    SkillList.prototype.getSkillByIndex = function(skillIndex) {
+      if (skillIndex < 0 || skillIndex >= this.skillsList.length) {
+        return null;
+      }
+      return this.skillsList[skillIndex];
+    };
+
+    // Returns all of the skill names in the list.
+    SkillList.prototype.getSkills = function() {
+      return this.skillsList;
+    };
+
+    // Returns the index for the given skill, based on a case-insensitive search
+    // of the skill list, or -1 if the skill name is not contained within the
+    // list.
+    SkillList.prototype.indexOfSkill = function(skillName) {
+      var lowercaseSkillName = skillName.toLowerCase();
+      for (var i = 0; i < this.skillsList.length; i++) {
+        if (this.skillsList[i] == lowercaseSkillName) {
+          return i;
+        }
+      }
+      return -1;
+    };
+
+    // Removes a skill by index from this list. Returns whether the skill was
+    // successfully removed, where the removal will fail if the index is out of
+    // bounds of the list.
+    SkillList.prototype.removeSkillByIndex = function(skillIndex) {
+      if (skillIndex < 0 || skillIndex >= this.skillsList.length) {
+        return false;
+      }
+      this.skillsList.splice(skillIndex, 1);
+      return true;
+    };
+
+    // Removes a skill based on a search name. Returns whether or not this was
+    // successful, which depends on whether the given skill name is actually in
+    // the skills list.
+    SkillList.prototype.removeSkillByName = function(skillName) {
+      var index = this.indexOfSkill(skillName);
+      return (index >= 0) && this.removeSkillByIndex(index);
+    };
+
+    // Returns the number of skills contained in this list.
+    SkillList.prototype.getSkillCount = function() {
+      return this.skillsList.length;
+    };
+
+    // Appends another skill list to this list, where duplicate skills are
+    // ignored.
+    SkillList.prototype.concatSkillList = function(otherSkillList) {
```

Comment:
I lean toward keeping it as ``concat``. It's concatenating a skill list (object) with the current object. ``addSkills`` and ``setSkills`` take arrays of skills, not a skill list domain object. I clarified the documentation further. Do you have any suggestions for a better name? I lean against ``addSkills`` because it makes more sense to be in the context of skill list.

Perhaps ``addSkillList``?